share/hybrid/71-u-d-c-gpu-detection.rules

```diff
@@ -3,7 +3,9 @@
 # Remove SimpleDRM device when nvidia-drm loads.
 # This normally happens automatically for DRM devices that also register
 # a framebuffer device, but that's not the case yet for the nvidia driver.
-ACTION=="add", SUBSYSTEM=="module", KERNEL=="nvidia_drm", TEST=="/sys/devices/platform/simple-framebuffer.0/drm/card0", RUN+="/bin/rm /dev/dri/card0"
+ACTION=="add", KERNEL=="card0", SUBSYSTEM=="drm", DRIVERS=="simple-framebuffer", ENV{U_D_C_IS_SIMPLEDRM}="1", RUN+="/bin/touch /run/u-d-c-card0-is-simpledrm"
+ACTION=="remove", KERNEL=="card0", SUBSYSTEM=="drm", ENV{U_D_C_IS_SIMPLEDRM}=="1", RUN+="/bin/rm -f /run/u-d-c-card0-is-simpledrm"
```

Comment:
`ACTION=="remove"`... doesn't sound testable right now. I imagine you would need a different kernel config where simpledrm is a module to test that. But for completeness it probably makes sense.

Reply:
This is not the event for module loaded, but for device add/remove, so it is testable.
I actually expected _remove_ to trigger every time i915 or amdgpu loads removing `/dev/dri/card0`, but apparently the whole sequence normally happens fast enough that udev isn't notified of card0 add nor remove.

However I tested it simulating an extra-slow module load time:
* append `modprobe.blacklist=amdgpu` to the kernel cmdline
* boot to a tty
* verify `/run/u-d-c-card0-is-simpledrm` exists
* `modprobe amdgpu`
* verify `/run/u-d-c-card0-is-simpledrm` was removed

The remove part is important because on multi-gpu systems after SimpleDRM is removed the card0 name would then be available for other GPUs to register and we need to make sure we're not removing those.

Reply:
It sounds like you're right, I'm just nervous about adding complexity. At least it will be confined to 24.10 until it's had time to mature before any backports...

This needs testing by at least one more person. If not me in the next few days then anyone else who can run Nvidia (ideally testing both single GPU and hybrid GPUs).

Reply:
> ideally testing both single GPU and hybrid GPUs

Right, hybrids is not something I can test